### PR TITLE
Azure patches for branch-5.0

### LINKED ISSUE #353: Error in post_start_script.py on Azure
I've been looking at logs when testing Scylla on Azure and got into such error:
```
< t:2022-05-06 19:56:43,308 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO | 2022-05-06 17:56:42,931 - [scylla_post_start] - WARNING - Error getting user data: 'azure_instance' object has no attribute 'user_data'
< t:2022-05-06 19:56:43,308 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO | Error getting user data: 'azure_instance' object has no attribute 'user_data'
< t:2022-05-06 19:56:43,308 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO | Traceback (most recent call last):
< t:2022-05-06 19:56:43,308 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO |   File "/opt/scylladb/scylla-machine-image/libexec/scylla_post_start.py", line 71, in <module>
< t:2022-05-06 19:56:43,308 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO |     if init.run_post_start_script():
< t:2022-05-06 19:56:43,409 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO |   File "/opt/scylladb/scylla-machine-image/libexec/scylla_post_start.py", line 57, in run_post_start_script
< t:2022-05-06 19:56:43,409 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO |     post_start_script = self.instance_user_data.get("post_start_script")
< t:2022-05-06 19:56:43,409 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !    INFO | AttributeError: 'NoneType' object has no attribute 'get'
< t:2022-05-06 19:56:43,409 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !  NOTICE | scylla-image-post-start.service: Main process exited, code=exited, status=1/FAILURE
< t:2022-05-06 19:56:43,410 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 ! WARNING | scylla-image-post-start.service: Failed with result 'exit-code'.
< t:2022-05-06 19:56:43,410 f:db_log_reader.py l:113  c:sdcm.db_log_reader   p:DEBUG > 2022-05-06T17:56:42+00:00 longevity-mini-test-ls-db-node-eastus-2 !     ERR | Failed to start Scylla cloud image post start service.

```

For now I don't have details about precise versions (recent master) or more logs. Let me know if you need it.

### PR DESCRIPTION
**Pull request by syuu1228**
Backporting Azure patches to branch-5.0.
It incudes following patches:
- [azure_instance: implement user-data](https://github.com/scylladb/scylla-machine-image/commit/37768f6b5d8ad40e26ef86a1db3fe03e56adc668)
- [azure_instance: fix metadata url for location and zone](https://github.com/scylladb/scylla-machine-image/commit/f2745ab01c48d969c6a9dabec4e6a19762c53271)
- [azure_instance: remove unneeded is_unsupported_instance_class()](https://github.com/scylladb/scylla-machine-image/commit/5b53ee3922f2f992d5f80d010a8b1cad6055e4b0)
- [azure_instance: fix bromen is_recommended_instance_size()](https://github.com/scylladb/scylla-machine-image/commit/c7b90846532ea484ceeaa581fb773d2935c31ce2)
- [azure_instance: support "swap disk"](https://github.com/scylladb/scylla-machine-image/commit/5d2c3eef957b0e9beff74ca1995cce86425fabdd)
- [azure_instance: fix failing is_azure_instance()](https://github.com/scylladb/scylla-machine-image/commit/eba6b6585906c14765eee497a1ba70ea54a15d7a)

Fixes #353

### CODE REVIEW

**Comment by syuu1228** [comment]
Jenkins test passed: https://jenkins.scylladb.com/view/scylla-5.0/job/scylla-5.0/job/releng-testing/job/byo/job/byo-package/1/

**Comment by benipeled** [comment]
> Jenkins test passed: https://jenkins.scylladb.com/view/scylla-5.0/job/scylla-5.0/job/releng-testing/job/byo/job/byo-package/1/

AFAIK the byo job builds the image but doesn't test it, so just to be on the safe side - please verify we can start a scylla-node with the built image.

File: lib/scylla_cloud.py [hunk 1]
```diff
@@ -500,6 +512,14 @@ def io_setup():
         return run('/opt/scylladb/scylla-machine-image/scylla_cloud_io_setup', shell=True, check=True)
 
 class aws_instance:
+    @property
+    def user_data(self):
+        encoded_user_data = self.__instance_metadata("/compute/userData")
+        if not encoded_user_data:
+            return ''
+        return base64.b64decode(encoded_user_data).decode()
```
Comment (soyacz) [inline lib/scylla_cloud.py:520]: This looks like it should be in azure class, shouldn't it?

**Comment by syuu1228** [comment]
> > Jenkins test passed: https://jenkins.scylladb.com/view/scylla-5.0/job/scylla-5.0/job/releng-testing/job/byo/job/byo-package/1/
> 
> AFAIK the byo job builds the image but doesn't test it, so just to be on the safe side - please verify we can start a scylla-node with the built image. (include user-data)

Oh okay, I will run test on it.

**Comment by syuu1228** [comment]
I able to manually launch the built Azure image on Ls instance type, it just able to start scylla-server without error.
But on Artifact tests for Azure image, it's keep failing before launching the instance.
Maybe it's related with getting "insufficient quota" error on Ls instance type which we recently facing?

@yaronkaikov Do you have any idea with this?

error message:
```
18:03:30  < t:2022-05-22 09:03:29,879 f:resource_group_provider.py l:48   c:sdcm.provision.azure.resource_group_provider p:INFO  > Creating SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus SCT resource group in region eastus...
18:03:30  < t:2022-05-22 09:03:30,070 f:resource_group_provider.py l:55   c:sdcm.provision.azure.resource_group_provider p:INFO  > Provisioned resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus in the eastus region
18:03:30  < t:2022-05-22 09:03:30,071 f:network_security_group_provider.py l:74   c:sdcm.provision.azure.network_security_group_provider p:INFO  > Creating SCT network security group in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:03:34  < t:2022-05-22 09:03:33,922 f:network_security_group_provider.py l:85   c:sdcm.provision.azure.network_security_group_provider p:INFO  > Provisioned security group default in the SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus resource group
18:03:34  < t:2022-05-22 09:03:33,927 f:virtual_network_provider.py l:47   c:sdcm.provision.azure.virtual_network_provider p:INFO  > Creating vnet in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:03:37  < t:2022-05-22 09:03:37,477 f:virtual_network_provider.py l:59   c:sdcm.provision.azure.virtual_network_provider p:INFO  > Provisioned vnet default in the SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus resource group
18:03:37  < t:2022-05-22 09:03:37,477 f:subnet_provider.py l:48   c:sdcm.provision.azure.subnet_provider p:INFO  > Creating subnet in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:03:41  < t:2022-05-22 09:03:40,971 f:subnet_provider.py l:61   c:sdcm.provision.azure.subnet_provider p:INFO  > Provisioned subnet default for the default vnet
18:03:41  < t:2022-05-22 09:03:40,971 f:ip_provider.py  l:48   c:sdcm.provision.azure.ip_provider p:INFO  > Creating public_ip in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:03:43  < t:2022-05-22 09:03:42,854 f:ip_provider.py  l:63   c:sdcm.provision.azure.ip_provider p:INFO  > Provisioned public ip artifacts-azure-image-jenkins-db-node-eastus-1-ipv4 (SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus) in the 20.231.80.138 resource group
18:03:43  < t:2022-05-22 09:03:42,855 f:network_interface_provider.py l:66   c:sdcm.provision.azure.network_interface_provider p:INFO  > Creating nic in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:04:13  < t:2022-05-22 09:04:13,623 f:network_interface_provider.py l:73   c:sdcm.provision.azure.network_interface_provider p:INFO  > Provisioned nic artifacts-azure-image-jenkins-db-node-eastus-1-nic in the SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus resource group
18:04:13  < t:2022-05-22 09:04:13,624 f:virtual_machine_provider.py l:50   c:sdcm.provision.azure.virtual_machine_provider p:INFO  > Creating 'artifacts-azure-image-jenkins-db-node-eastus-1' VM in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:04:13  < t:2022-05-22 09:04:13,624 f:virtual_machine_provider.py l:52   c:sdcm.provision.azure.virtual_machine_provider p:INFO  > Instance params: InstanceDefinition(name='artifacts-azure-image-jenkins-db-node-eastus-1', image_id='debug-image-ScyllaDB-5.1.dev-0.20220519.f873806c7c98-1-build-2', type='Standard_L8s_v2', user_name='scyllaadm', tags={'RunByUser': 'syuu', 'TestName': 'artifacts_test.ArtifactsTest.test_scylla_service', 'TestId': 'a503cc28-54f9-48a0-9f78-f3bdf829b003', 'version': 'scylla-5.0', 'JenkinsJobTag': 'jenkins-scylla-5.0-releng-testing-artifacts-artifacts-azure-image-test-6', 'NodeType': 'scylla-db', 'keep_action': 'terminate'}, arch=<VmArch.X86: 'x86_64'>, root_disk_size=None, data_disks=None)
18:04:15  < t:2022-05-22 09:04:14,627 f:virtual_machine_provider.py l:50   c:sdcm.provision.azure.virtual_machine_provider p:INFO  > Creating 'artifacts-azure-image-jenkins-db-node-eastus-1' VM in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:04:15  < t:2022-05-22 09:04:14,627 f:virtual_machine_provider.py l:52   c:sdcm.provision.azure.virtual_machine_provider p:INFO  > Instance params: InstanceDefinition(name='artifacts-azure-image-jenkins-db-node-eastus-1', image_id='debug-image-ScyllaDB-5.1.dev-0.20220519.f873806c7c98-1-build-2', type='Standard_L8s_v2', user_name='scyllaadm', tags={'RunByUser': 'syuu', 'TestName': 'artifacts_test.ArtifactsTest.test_scylla_service', 'TestId': 'a503cc28-54f9-48a0-9f78-f3bdf829b003', 'version': 'scylla-5.0', 'JenkinsJobTag': 'jenkins-scylla-5.0-releng-testing-artifacts-artifacts-azure-image-test-6', 'NodeType': 'scylla-db', 'keep_action': 'terminate'}, arch=<VmArch.X86: 'x86_64'>, root_disk_size=None, data_disks=None)
18:04:16  < t:2022-05-22 09:04:15,629 f:virtual_machine_provider.py l:50   c:sdcm.provision.azure.virtual_machine_provider p:INFO  > Creating 'artifacts-azure-image-jenkins-db-node-eastus-1' VM in resource group SCT-a503cc28-54f9-48a0-9f78-f3bdf829b003-eastus...
18:04:16  < t:2022-05-22 09:04:15,630 f:virtual_machine_provider.py l:52   c:sdcm.provision.azure.virtual_machine_provider p:INFO  > Instance params: InstanceDefinition(name='artifacts-azure-image-jenkins-db-node-eastus-1', image_id='debug-image-ScyllaDB-5.1.dev-0.20220519.f873806c7c98-1-build-2', type='Standard_L8s_v2', user_name='scyllaadm', tags={'RunByUser': 'syuu', 'TestName': 'artifacts_test.ArtifactsTest.test_scylla_service', 'TestId': 'a503cc28-54f9-48a0-9f78-f3bdf829b003', 'version': 'scylla-5.0', 'JenkinsJobTag': 'jenkins-scylla-5.0-releng-testing-artifacts-artifacts-azure-image-test-6', 'NodeType': 'scylla-db', 'keep_action': 'terminate'}, arch=<VmArch.X86: 'x86_64'>, root_disk_size=None, data_disks=None)
18:04:17  < t:2022-05-22 09:04:16,631 f:decorators.py   l:69   c:sdcm.utils.decorators p:ERROR > '_provision_instance': Number of retries exceeded!
18:04:17  < t:2022-05-22 09:04:16,634 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2022-05-22 09:04:16.632: (TestFrameworkEvent Severity.ERROR) period_type=one-time event_id=e72f2250-7c49-4c62-bf58-eba13a79274c, source=ArtifactsTest.SetUp()
18:04:17  < t:2022-05-22 09:04:16,634 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > exception=list index out of range
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR > Exception in setUp. Will call tearDown < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR > Exception in setUp. Will call tearDown
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR > Traceback (most recent call last):
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/tester.py", line 133, in wrapper
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     return method(*args, **kwargs)
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/utils/decorators.py", line 113, in inner
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     res = func(*args, **kwargs)
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/tester.py", line 616, in setUp
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     self.init_resources()
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/tester.py", line 1426, in init_resources
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     self.get_cluster_azure(loader_info=loader_info, db_info=db_info,
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/tester.py", line 883, in get_cluster_azure
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     self.db_cluster = ScyllaAzureCluster(image_id=azure_image,
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/cluster_azure.py", line 230, in __init__
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     super().__init__(
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/cluster.py", line 3586, in __init__
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     super().__init__(*args, **kwargs)
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/cluster_azure.py", line 142, in __init__
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     super().__init__(cluster_uuid=cluster_uuid,
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/cluster.py", line 3085, in __init__
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     self.add_nodes(num, dc_idx=dc_idx, enable_auto_bootstrap=self.auto_bootstrap)
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/cluster_azure.py", line 155, in add_nodes
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     instances = self._create_instances(count, dc_idx)
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/cluster_azure.py", line 198, in _create_instances
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     instances.append(self._provision_instance(instance_definition=instance_definition,
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/utils/decorators.py", line 64, in inner
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     return func(*args, **kwargs)
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/cluster_azure.py", line 204, in _provision_instance
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     return self.provisioners[dc_idx].get_or_create_instance(definition=instance_definition,
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/provision/azure/provisioner.py", line 86, in get_or_create_instance
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     v_m = self._vm_provider.get_or_create(definition, nic_id, pricing_model)
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/provision/azure/virtual_machine_provider.py", line 74, in get_or_create
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     storage_profile = self._get_scylla_storage_profile(image_id=definition.image_id, name=definition.name,
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/provision/azure/virtual_machine_provider.py", line 181, in _get_scylla_storage_profile
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR >     "offer": image_reference_values[1],
18:04:17  < t:2022-05-22 09:04:16,632 f:tester.py       l:140  c:sdcm.tester          p:ERROR > IndexError: list index out of range
18:04:17  < t:2022-05-22 09:04:16,639 f:tester.py       l:2403 c:ArtifactsTest        p:INFO  > TearDown is starting...
```
Failed test:
https://jenkins.scylladb.com/job/scylla-5.0/job/releng-testing/job/artifacts/job/artifacts-azure-image-test/6/

**Comment by fgelcer** [comment]
@soyacz , who populates this `"offer": image_reference_values[1]` ?

**Comment by syuu1228** [comment]
Hmm, even after I specified full path of Azure image id, the test still failing.
Looks like we become able to start instance with correct id, I even verify it in Azure console, but the test is failing at connecting the instance via SSH.
Looks like it just timed out, I able to find these errors multiple times on the log:
```
20:07:23  < t:2022-05-23 11:07:22,945 f:remote_libssh_cmd_runner.py l:66   c:RemoteLibSSH2CmdRunner p:ERROR > Exception:  File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/remote/libssh2_client/__init__.py", line 589, in run
20:07:23  < t:2022-05-23 11:07:22,945 f:remote_libssh_cmd_runner.py l:66   c:RemoteLibSSH2CmdRunner p:ERROR >     channel = self.open_channel()
20:07:23  < t:2022-05-23 11:07:22,945 f:remote_libssh_cmd_runner.py l:66   c:RemoteLibSSH2CmdRunner p:ERROR >   File "/home/jenkins/slave/workspace/scylla-5.0/releng-testing/artifacts/artifacts-azure-image-test/scylla-cluster-tests/sdcm/remote/libssh2_client/__init__.py", line 687, in open_channel
20:07:23  < t:2022-05-23 11:07:22,945 f:remote_libssh_cmd_runner.py l:66   c:RemoteLibSSH2CmdRunner p:ERROR >     raise OpenChannelTimeout(f'Failed to open channel in {timeout} seconds')
20:07:23  < t:2022-05-23 11:07:22,945 f:remote_libssh_cmd_runner.py l:66   c:RemoteLibSSH2CmdRunner p:ERROR > 
20:07:23  < t:2022-05-23 11:07:22,945 f:remote_libssh_cmd_runner.py l:66   c:RemoteLibSSH2CmdRunner p:ERROR > Failed to open channel in 15 seconds
20:09:59  < t:2022-05-23 11:09:56,291 f:remote_libssh_cmd_runner.py l:66   c:RemoteLibSSH2CmdRunner p:ERROR > Failed to run a command due to exception!
```
 https://jenkins.scylladb.com/job/scylla-5.0/job/releng-testing/job/artifacts/job/artifacts-azure-image-test/7/consoleFull

**Comment by soyacz** [comment]
I'm looking into it. Looks not related to your changes.

**Comment by soyacz** [comment]
@syuu1228 when I tested it from my host it works and test passes. Somehow it is having some ssh problems when using jenkins builder. I'll investigate it.

**Comment by syuu1228** [comment]
> @syuu1228 when I tested it from my host it works and test passes. Somehow it is having some ssh problems when using jenkins builder. I'll investigate it.

Can we merge this now, since both my manual test and the test on @soyacz's environment passed, and very likely the error on jenkins is not related with this patchset?

File: lib/scylla_cloud.py (hunk 1 above)
Comment (fgelcer) [reply lib/scylla_cloud.py:519]: @syuu1228 , was this fixed? i don't see any changes with this change

**Review by fgelcer — CHANGES_REQUESTED** [review]

File: lib/scylla_cloud.py (hunk 1 above)
Comment (syuu1228) [reply lib/scylla_cloud.py:519]: @fgelcer it fixed, see the change https://github.com/scylladb/scylla-machine-image/compare/7ee809bb2423717b025d226f2d8a413836fa7f84..7df851bd64107b2967f3fb54842bd049fe555ac2

Comment (fgelcer) [reply lib/scylla_cloud.py:519]: great, thank you...
in that case, LGTM

**Review by soyacz — APPROVED** [review]

**Review by benipeled — APPROVED** [review]

**Review by yaronkaikov — APPROVED** [review]

**Pull request merged**